clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java

```diff
@@ -583,12 +579,15 @@ private boolean deleteHelper(ObjectLocation loc) throws IOException {
 
     /**
      * Delete parents directory markers from path until root.
-     * Assume the caller created an object under the path which will make the empty directory irrelevant.
+     * Assume the caller created an object under the path which will make the empty
+     * directory irrelevant.
      * Based on the S3AFileSystem implementation.
-     * NOTE there is a race with mkdir which in case we move a file to a directory which mkdirs try to create, in case we try to delete
+     * NOTE there is a race with mkdir which in case we move a file to a directory
+     * which mkdirs try to create, in case we try to delete
+     * 
      * @param f path to start for empty directory markers
      */
-    void deleteEmptyDirectoryMarkers(Path f) {
+    public void deleteEmptyDirectoryMarkers(Path f) {
```

Comment:
I created the "storage" package. Classes from this package use this method.

Reply:
We need to find a way to hide this from outside of the Hadoop classes.  Otherwise it will become a part of the external interface, and we'll be stuck.  (Hadoop does something similar with annotations, but it's probably too heavyweight for us here.)

For velocity I suggest:
* Add a comment not to call this from outside "our" packages.
* Open an issue to hide this into "our" packages.